pkg/commands/tls/custom/certificate/certificate_test.go

```diff
@@ -52,6 +55,28 @@ func TestTLSCustomCertCreate(t *testing.T) {
 			Args:       args("tls-custom certificate create --cert-blob example"),
 			WantOutput: fmt.Sprintf("Created TLS Certificate '%s'", mockResponseID),
 		},
+		{
+			Name: validateFilePathFailure,
+			API: mock.API{
+				CreateCustomTLSCertificateFn: func(_ *fastly.CreateCustomTLSCertificateInput) (*fastly.CustomTLSCertificate, error) {
+					return nil, testutil.Err
+				},
+			},
+			Args:      args(fmt.Sprintf("tls-custom certificate create --cert-blob %s", filepath.Join("invalid", "path", "missing.crt"))),
+			WantError: testutil.Err.Error(),
+		},
+		{
+			Name: validateFilePathSuccess,
+			API: mock.API{
+				CreateCustomTLSCertificateFn: func(_ *fastly.CreateCustomTLSCertificateInput) (*fastly.CustomTLSCertificate, error) {
+					return &fastly.CustomTLSCertificate{
+						ID: mockResponseID,
+					}, nil
+				},
+			},
+			Args:       args(fmt.Sprintf("tls-custom certificate create --cert-blob %s", filepath.Join("pkg", "config", "testdata", "certificate.crt"))),
```

Comment:
I'm not sure how this is passing because from what I can tell there isn't a file at `pkg/config/testdata/certificate.crt`.

Reply:
Uh yep that is not good. I forgot to commit that file, but it should fail in CI.

Reply:
Yeah let's make sure ci fails first then commit the file to see if it passes. You can of course run the tests locally to help debug things

Reply:
It's failing b/c if `argparser.Content` can't find the file it thinks the argument is the content. So the cert-blob being uploaded is `pkg/config/testdata/certificate.crt` . A filename passed in that doesn't exist becomes the raw cert blob contents. Will think about how to handle this.

Reply:
https://github.com/fastly/cli/blob/46fefb53dac87aaef3898c9d1094baac4281f26c/pkg/argparser/flags.go#L276

Reply:
So the question is, do we do more heuristics or validation on the cert-blob before uploading to check if we're uploading a cert? Or do we do some heuristics on if the cert-blob looks like a file path? 

My concern is if someone gives an invalid path, the client will upload a 'cert' that's invalid. They then get this error:

```
➜  cli git:(rdoherty/certificate-create-accept-file-path) ✗ ./fastly tls-custom certificate create --cert-blob=bad/path

ERROR: the Fastly API returned 400 Bad Request: Invalid value for cert_blob (The contents provided do not appear to be a valid X.509 certificate.).

```

Which isn't too helpful. Only if they use `--debug-mode` would they see that the 'contents' are the file path. We could use `pem.Decode()` and if it fails return an error. Thoughts?

Reply:
OK, I think we need to start handling the error case in `argparser.Content`...

https://github.com/fastly/cli/blob/main/pkg/argparser/flags.go#L281C43-L281C43

...so this ^^ should change to `!= nil` rather than the happy path `== nil`.

e.g.

```go
func Content(flagval string) (string, error) {
	content := flagval
	if path, err := filepath.Abs(flagval); err == nil {
		if _, err := os.Stat(path); err != nil {
			return fmt.Errorf("error looking up provided path '%s': %q", path, err)
		}
		if data, err := os.ReadFile(path); err == nil /* #nosec */ {
			content = string(data)
		}
	}
	return content
}
```

Notice the function signature has changed to returning two arguments now `(string, error)`.

This will mean updating the callers to handle an error now as well. I think it's only called in like 5 places in the codebase though so shouldn't be a massive amount of work.

Well done spotting this bug 👍🏻 🙂